.github/workflows/multi-approvers.js

```diff
@@ -2,49 +2,120 @@ const APPROVED = 'APPROVED';
+    const reviewStateByLogin = {};
+    submittedReviews
+      // Remove the PR user.
+      .filter((r) => r.user.login !== prLogin)
+      // Only consider users in the org.
+      .filter((r) => this.contains(r.user.login))
+      // Sort chronologically ascending. Note that a reviewer can submit multiple reviews.
+      .sort((a, b) => new Date(a.submitted_at) - new Date(b.submitted_at))
+      .forEach((r) => {
+        const reviewerLogin = r.user.login;
+
+        // Set state if it does not exist.
+        if (!Object.hasOwn(reviewStateByLogin, reviewerLogin)) {
+          reviewStateByLogin[reviewerLogin] = r.state;
+          return;
+        }
+
+        // Always update state if not approved.
+        if (reviewStateByLogin[reviewerLogin] !== APPROVED) {
+          reviewStateByLogin[reviewerLogin] = r.state;
+          return;
+        }
+
+        // Do not update approved state for a comment.
+        if (reviewStateByLogin[reviewerLogin] === APPROVED && r.state !== COMMENTED) {
+          reviewStateByLogin[reviewerLogin] = r.state;
+        }
+      })
+
+    return Object.values(reviewStateByLogin).filter((s) => s === APPROVED).length;
+  }
+}
 
-      // Always update state if not approved.
-      if (reviewStateByLogin[reviewerLogin] !== APPROVED) {
-        reviewStateByLogin[reviewerLogin] = r.state;
-        return;
-      }
+/** Members backed by a JSON file. */
+class JsonMembers extends AbstractMembers {
+  members;
 
-      // Do not update approved state for a comment.
-      if (reviewStateByLogin[reviewerLogin] === APPROVED && r.state !== COMMENTED) {
-        reviewStateByLogin[reviewerLogin] = r.state;
-      }
-    })
+  constructor(membersPath) {
+    super();
+    this.members = require(membersPath).reduce((acc, v) => acc.set(v.login, v), new Map());
+  }
 
-  return Object.values(reviewStateByLogin).filter((s) => s === APPROVED).length;
+  async contains(login) {
+    return this.members.has(login);
+  }
+}
+
+/** Members backed by a GitHub team. */
+class TeamMembers extends AbstractMembers {
+  static ALLOWED_ROLES = ["maintainer", "member"];
+  static ACTIVE = "active";
+
+  org;
+  teamSlug;
+  github;
+
+  constructor(org, teamSlug, github) {
+    super();
+    this.org = org;
+    this.teamSlug = teamSlug;
+    this.github = github;
+  }
+
+  async contains(login) {
+    try {
+      const response = await this.github.rest.teams.getMembershipForUserInOrg({
+        org: this.org,
+        team_slug: this.teamSlug,
+        username: login,
+      });
+      return TeamMembers.ALLOWED_ROLES.indexOf(response.data.role) >= 0 &&
+          response.data.state === TeamMembers.ACTIVE;
+    } catch (error) {
+      if (error.status === 404) {
+        // We can get here for two reasons:
+        // 1) The user is not a member
+        // 2) The team does not exist
+        // Either way, it's safe to return false here.
+        return false;
+      }
+      throw error;
+    }
```

Comment:
We should really log the error (`this.core.debug`) or something.

Reply:
Will address this in a different PR where I will use an Action instead of a Workflow.

Reply:
Please create a follow-up issue.